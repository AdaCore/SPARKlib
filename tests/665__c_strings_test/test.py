from test_support import prove_all, sparklib_exec_test
import os

os.environ["SPARKLIB_BODY_MODE"] = "On"

if __name__ == "__main__":
    prove_all(sparklib=True, steps=1000, opt=["--no-inlining"])
    sparklib_exec_test("test.gpr", "./obj/test_string")
