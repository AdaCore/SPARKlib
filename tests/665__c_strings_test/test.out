test_string.adb:22:10: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:22:27: info: precondition proved (CVC5: 1 VC)
test_string.adb:24:10: info: precondition proved (Z3: 1 VC)
test_string.adb:24:18: info: precondition proved (CVC5: 1 VC)
test_string.adb:24:37: info: precondition proved (CVC5: 1 VC)
test_string.adb:31:10: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:31:27: info: precondition proved (CVC5: 1 VC)
test_string.adb:33:10: info: precondition proved (Z3: 1 VC)
test_string.adb:33:18: info: precondition proved (CVC5: 1 VC)
test_string.adb:33:37: info: precondition proved (CVC5: 1 VC)
test_string.adb:44:10: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:46:10: medium: precondition might fail
test_string.adb:46:18: info: precondition proved (CVC5: 1 VC)
test_string.adb:46:37: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test_string.adb:53:10: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:55:10: medium: precondition might fail
test_string.adb:55:18: info: precondition proved (CVC5: 1 VC)
test_string.adb:55:37: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test_string.adb:62:07: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:62:24: info: precondition proved (CVC5: 1 VC)
test_string.adb:65:07: info: precondition proved (CVC5: 1 VC)
test_string.adb:70:07: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:70:24: info: precondition proved (CVC5: 1 VC)
test_string.adb:71:25: info: precondition proved (CVC5: 1 VC)
test_string.adb:73:07: info: precondition proved (CVC5: 1 VC)
test_string.adb:74:07: info: precondition proved (Z3: 1 VC)
test_string.adb:80:07: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:80:24: info: precondition proved (CVC5: 1 VC)
test_string.adb:85:28: info: precondition proved (CVC5: 2 VC)
test_string.adb:87:10: info: precondition proved (CVC5: 1 VC)
test_string.adb:88:10: info: precondition proved (Z3: 1 VC)
test_string.adb:93:28: info: precondition proved (CVC5: 2 VC)
test_string.adb:95:10: info: precondition proved (CVC5: 1 VC)
test_string.adb:96:10: info: precondition proved (Z3: 1 VC)
test_string.adb:103:07: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:104:21: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test_string.adb:106:07: info: precondition proved (CVC5: 1 VC)
test_string.adb:107:07: medium: precondition might fail
test_string.adb:113:07: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:118:24: info: precondition proved (CVC5: 3 VC)
test_string.adb:120:10: info: precondition proved (CVC5: 1 VC)
test_string.adb:121:10: info: precondition proved (Z3: 1 VC)
test_string.adb:126:24: info: precondition proved (CVC5: 3 VC)
test_string.adb:128:10: info: precondition proved (CVC5: 1 VC)
test_string.adb:129:10: medium: precondition might fail
test_string.adb:136:07: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:136:24: info: precondition proved (CVC5: 1 VC)
test_string.adb:138:07: info: precondition proved (Z3: 1 VC)
test_string.adb:138:15: info: precondition proved (CVC5: 1 VC)
test_string.adb:148:10: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:148:27: info: precondition proved (CVC5: 1 VC)
test_string.adb:150:10: info: precondition proved (CVC5: 3 VC; Z3: 1 VC)
test_string.adb:151:10: info: precondition proved (Z3: 1 VC)
test_string.adb:151:18: info: precondition proved (CVC5: 1 VC)
test_string.adb:151:42: info: precondition proved (CVC5: 1 VC)
test_string.adb:159:10: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:159:27: info: precondition proved (CVC5: 1 VC)
test_string.adb:161:10: info: precondition proved (CVC5: 3 VC; Z3: 1 VC)
test_string.adb:162:10: info: precondition proved (Z3: 1 VC)
test_string.adb:162:18: info: precondition proved (CVC5: 1 VC)
test_string.adb:162:43: info: precondition proved (CVC5: 1 VC)
test_string.adb:174:10: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:176:10: info: precondition proved (CVC5: 2 VC; Z3: 1 VC)
test_string.adb:177:10: medium: precondition might fail
test_string.adb:177:18: info: precondition proved (CVC5: 1 VC)
test_string.adb:177:42: info: precondition proved (CVC5: 2 VC)
test_string.adb:185:10: info: absence of resource or memory leak at end of scope proved (CVC5: 1 VC)
test_string.adb:187:10: medium: precondition might fail, cannot prove Str'Length + Offset <= Strlen (Item)
test_string.adb:188:10: medium: precondition might fail
test_string.adb:188:18: info: precondition proved (CVC5: 1 VC)
test_string.adb:188:43: info: precondition proved (CVC5: 2 VC)
